### PR TITLE
fix(common): invalid ImageKit quality parameter

### PR DESCRIPTION
**Pull request by johnnygerard**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #55192

## What is the new behavior?
Placeholder URLs of this form:
https://ik.imagekit.io/demo/tr:w-30,q-20/medium_cafe_B1iTdD0C.jpg

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts [hunk 1]
```diff
@@ -45,20 +45,22 @@ export function createImagekitUrl(path: string, config: ImageLoaderConfig): stri
...
-
   // When requesting a placeholder image we ask for a low quality image to reduce the load time.
   if (config.isPlaceholder) {
-    url.searchParams.set('q', PLACEHOLDER_QUALITY);
+    params.push(`q-${PLACEHOLDER_QUALITY}`);
+  }
+
+  if (params.length) {
+    urlSegments.splice(1, 0, `tr:${params.join(',')}`);
   }
```
Comment (JeanMeche) [inline packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts:62]: wdyt of going with something like
```suggestion
const urlSergments = params.length ? [path, `tr:${params.join(',')}`, src] : [path, src]
```

Comment (johnnygerard) [reply packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts:62]: Yes, it is more readable.

File: packages/common/test/image_loaders/image_loader_spec.ts [hunk 2]
```diff
@@ -171,8 +171,8 @@ describe('Built-in image directive loaders', () => {
     it('should load a low quality image when a placeholder is requested', () => {
       const path = 'https://ik.imageengine.io/imagetest';
       const loader = createImageKitLoader(path);
-      const config = {src: 'img.png', isPlaceholder: true};
-      expect(loader(config)).toBe(`${path}/img.png?q=20`);
+      const config = {src: 'img.png', isPlaceholder: true, width: 30};
+      expect(loader(config)).toBe(`${path}/tr:w-30,q-20/img.png`);
```
[the remark this answers is not in the document]
Comment (johnnygerard) [reply packages/common/test/image_loaders/image_loader_spec.ts:175]: Is it better a separate test or an extra expectation?

[the remark this answers is not in the document]
Comment (JeanMeche) [reply packages/common/test/image_loaders/image_loader_spec.ts:180]: I'd be fine with having it in the same expectation, since its a pretty simple test.

**Review by JeanMeche — APPROVED** [review]
LGTM 
Thank you for catching the issue and providing a fix !

**Review by AndrewKushnir — APPROVED** [review]

**Comment by thePunderWoman** [comment]
This PR was merged into the repository by commit 58e2f47426e1d2a1351e6b0721adf438d5ac33e8.

**Pull request closed without merging**